airbyte-integrations/connectors/source-flexport/source_flexport/source.py

```diff
@@ -0,0 +1,198 @@
+from urllib.parse import parse_qsl, urlparse
+
+import pendulum
+import requests
+from airbyte_cdk.sources import AbstractSource
+from airbyte_cdk.sources.streams import Stream
+from airbyte_cdk.sources.streams.http import HttpStream
+from airbyte_cdk.sources.streams.http.auth.core import HttpAuthenticator
+from airbyte_cdk.sources.streams.http.requests_native_auth import TokenAuthenticator
+from requests.auth import AuthBase
+
+
+class FlexportError(Exception):
+    pass
+
+
+class FlexportStream(HttpStream, ABC):
+    url_base = "https://api.flexport.com/"
+    raise_on_http_errors = False
+    primary_key = "id"
+    page_size = 500
+
+    def __init__(self, authenticator: Union[AuthBase, HttpAuthenticator] = None, start_date: str = None):
+        super().__init__(authenticator=authenticator)
+
+        self._authenticator = authenticator
+        self.start_date = start_date
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        # https://apidocs.flexport.com/reference/pagination
+        # All list endpoints return paginated responses. The response object contains
+        # elements of the current page, and links to the previous and next pages.
+        data = response.json()["data"]
+
+        if data["next"]:
+            url = urlparse(data["next"])
+            qs = dict(parse_qsl(url.query))
+
+            return {
+                "page": qs["page"],
+                "per": qs["per"],
+            }
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        if next_page_token:
+            return next_page_token
+
+        return {
+            "page": 1,
+            "per": self.page_size,
+        }
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        # https://apidocs.flexport.com/reference/response-layout
+        json = response.json()
+
+        http_error = None
+        try:
+            response.raise_for_status()
+        except Exception as exc:
+            http_error = exc
+
+        error = json.get("error")
+        if error:
+            if "code" in error and "message" in error:
+                raise FlexportError(f"{error['code']}: {error['message']}") from http_error
+
+            if not http_error:
+                raise FlexportError(f"Unexpected error: {error}")
+
+        if http_error:
+            raise http_error
+
+        yield from json["data"]["data"]
+
+
+class IncrementalFlexportStream(FlexportStream, ABC):
+    @property
+    def cursor_field(self) -> str:
+        return []
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        current = current_stream_state.get(self.cursor_field, "")
```

Comment:
```suggestion
        current = current_stream_state.get(self.cursor_field, 0)
```
Because you're using `max` I'd recommend using an int for comparison.

Reply:
The cursor value is ISO 8601 formatted string. Comparison with `int` will raise:

```python
>>> max(datetime.datetime.today().isoformat(), 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'int' and 'str'
```

Reply:
nit: you could set `1970-01-01 00:00:00.000.`

Reply:
Fixed in 0f45987dfe60b2e9421ad2d7697bebdbad6c267f.